src/rgw/driver/rados/rgw_sal_rados.cc

```diff
@@ -3073,8 +3073,18 @@ int RadosObject::set_cloud_restore_status(const DoutPrefixProvider* dpp,
   bufferlist bl;
   using ceph::encode;
   encode(restore_status, bl);
-
-  ret = modify_obj_attrs(RGW_ATTR_RESTORE_STATUS, bl, y, dpp);
+  rgw_obj target = get_obj();
```

Comment:
Instead of repeating this code here, can you please add extra arg "log_op (default true)" to modify_obj_attrs() similar to write_meta() and set it to "false"